docs/zkapps/zkapp-development-frameworks.mdx

```diff
@@ -15,68 +15,214 @@ keywords:
   - protokit
 ---
 
-:::info
 
-To protect end users and ensure your zkApps are secure, consider the information at [Security and zkApps](/zkapps/writing-a-zkapp/introduction-to-zkapps/secure-zkapps) while architecting your solution and consider a third-party security audit before deploying to Mina mainnet.
+#  zkApp Development Frameworks
 
-:::
+Applications on Mina are called zkApps, and they come in a wide range of forms, including DeFi, games, state bridges, and more. However, due to Mina’s client execution model, the developer experience differs somewhat from that of other blockchain networks, such as Ethereum.
 
-# zkApp Development Frameworks
+Developers build zkApps with **o1js**. Developers can either use o1js directly or use o1js with **Protokit**, a framework built on top of o1js. Both options offer the full benefits of Mina’s verifiable, privacy-preserving zero knowledge proofs, but offer different tradeoffs between developer experience, performance, and decentralization.
```

Comment:
> Both options offer the full benefits of Mina’s verifiable, privacy-preserving zero knowledge proofs

Do they actually? Natively, afaik Protokit executes business logic on a server/"on-chain" - away from the client - this would render __privacy-preserving__ not true in the default case

Reply:
An example from the [protokit docs](https://protokit.dev/docs/runtime/methods#proofs-as-arguments):

```ts
@runtimeModule()
export class Balances extends RuntimeModule<Record<string, never>> {
  @state() public balances = StateMap.from(PublicKey, UInt64);
 
  @runtimeMethod()
  public mint(canMintProof: CanMintProof, amount: UInt64) {
    canMintProof.verify();
    this.balances.set(address, amount);
  }
}
```

Seems like using a proof input would be a valid way of preserving privacy on protokit.

Reply:
That's not the default tho whereas with o1js it is, because everything happens locally on the clients machine

Reply:
There's a difference between opt-in and opt-out

Reply:
>  So privacy is indeed optional, but that's actually a feature you can leverage, which mitigates the concurrency issues of current smart contracts.

I am not saying it isn't a feature! But a big difference that users should be made aware of, especially since the default is on-chain execution.

Reply:
Is everyone ok to leave this as is? The statement is still correct - both frameworks allow for privacy. The details of how to implement for privacy can be left for deeper level docs. The key point to land here is that you can build a privacy-preserving app using either framework.